TM.cpp

```diff
@@ -174,4 +174,17 @@ bool TM::operator==(TM const& rhs)
     }
 
     return true;
-}
\ No newline at end of file
+}
+
+friend std::ostream& operator<<(std::ostream& strm, TM const& tm)
+{
+    strm << "Start State: " << *tm.startState.getName() << "\n"
+    for (auto const& [key1, state] : state_map)
+    {
+      strm << "\t" << "State Name: " <<  state.name << "\n" << "\t" << "State Type: " << state.type << "\n"
+      for (auto const& [key2, trans] : *state.trans_map)
+      {
+        strm << "\t\t" << trans << "\n";
```

Comment:
i still dont get what the read symbol is and shouldnt i want to output transition the same way i would print a singular transition

Reply:
oh... so the transition overloaded << will never be able to know what its read symbol is. will this cause issues in the future?

Reply:
I don't think it will cause issues since we are only printing transitions for our own use.

But, we could just add another member variable to the `Transition` class that stores the read symbol. Then, the overloaded operator << for `Transition` would print everything correctly, including the correct read symbol (instead of simply printing the string `read`).

Reply:
Also, I graphed the simple TM for adding 1 to a binary number: https://app.lucidchart.com/invitations/accept/18d03b62-7b94-47e3-acb8-043ea916363f. The read symbols are the 0's, 1's, and _'s that come before the `->` in the transition definitions. (I used '_' to mean the blank symbol, and I am using the model of the two-way TM where blank symbols extend to the left and right of the input string.)